File: add-suggestion.js
import { google } from 'googleapis';

export default async function handler(req, res) {
  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  try {
    const auth = new google.auth.GoogleAuth({
      credentials: JSON.parse(process.env.GOOGLE_SERVICE_ACCOUNT),
      scopes: ['https://www.googleapis.com/auth/spreadsheets'],
    });

    const sheets = google.sheets({ version: 'v4', auth });
    const { nick, title, category, link, comment } = req.body;

    const now = new Date().toLocaleString('ru-RU');
    const suggestionId = Date.now().toString();

    await sheets.spreadsheets.values.append({
      spreadsheetId: process.env.SHEET_ID,
      range: 'Лист1!A:H',
      valueInputOption: 'USER_ENTERED',
      requestBody: {
        values: [[
          now,
          nick,
          title,
          category,
          link || '-',
          comment || '-',
          '0',
          suggestionId
        ]]
      }
    });

    res.setHeader('Access-Control-Allow-Origin', '*');
    res.status(200).json({ success: true, id: suggestionId });
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
}

File: add_suggestion.py
from google.oauth2 import service_account
from googleapiclient.discovery import build
import os
import json
from datetime import datetime
import uuid

def handler(request):
    if request.method != 'POST':
        return {
            'statusCode': 405,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps({'error': 'Method not allowed'})
        }
    
    try:
        # Получаем данные из запроса
        data = json.loads(request.body)
        
        # Настройки Google Sheets
        SCOPES = ['https://www.googleapis.com/auth/spreadsheets']
        SERVICE_ACCOUNT_INFO = json.loads(os.environ.get('GOOGLE_SERVICE_ACCOUNT'))
        SPREADSHEET_ID = os.environ.get('SHEET_ID')
        RANGE_NAME = 'Лист1!A:H'
        
        credentials = service_account.Credentials.from_service_account_info(
            SERVICE_ACCOUNT_INFO, scopes=SCOPES)
        service = build('sheets', 'v4', credentials=credentials)
        
        # Подготовка данных для записи
        now = datetime.now().strftime('%d.%m.%Y %H:%M')
        suggestion_id = str(uuid.uuid4())
        
        values = [[
            now,
            data.get('nick', 'Аноним'),
            data.get('title', ''),
            data.get('category', 'Другое'),
            data.get('link', '-'),
            data.get('comment', '-'),
            '0',
            suggestion_id
        ]]
        
        body = {
            'values': values
        }
        
        # Добавляем строку
        service.spreadsheets().values().append(
            spreadsheetId=SPREADSHEET_ID,
            range=RANGE_NAME,
            valueInputOption='USER_ENTERED',
            body=body
        ).execute()
        
        return {
            'statusCode': 200,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps({'success': True, 'id': suggestion_id})
        }
        
    except Exception as e:
        return {
            'statusCode': 500,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps({'error': str(e)})
        }